llm_studio/python_configs/text_dpo_modeling_config.py

```diff
@@ -74,7 +75,21 @@ class ConfigDPOTraining(ConfigNLPCausalLMTraining):
     def __post_init__(self):
         super().__post_init__()
         self._possible_values["beta"] = possible_values.Number(0.05, 1.0, 0.05)
+        self._possible_values["simpo_gamma"] = possible_values.Number(0.05, 2.0, 0.05)
+
+        self._grid_search_values["beta"] = (0.1, 0.15, 0.20, 0.25, 0.4, 0.5)
+        self._grid_search_values["simpo_gamma"] = (0.5, 0.75, 1, 1.25, 1.5, 1.75, 2)
+
+        self._grid_search_iscustom["beta"] = True
+        self._grid_search_iscustom["simpo_gamma"] = True
+
+        self._nesting.add(
+            ["simpo_gamma"],
+            [Dependency(key="loss_function", value="SimPOLoss", is_set=True)],
+        )
```

Comment:
That was something that I wanted to avoid when adding the grid search. Now, when starting a grid search with DPO + SPO and selecting SPO only grid search params, the warning message will return a wrong number of experiments being started. 
When actually starting the experiment, this is catched though, and duplicate experiments are not started. 

Maybe this is fine and kind of an edge case, but would be great at least open an issue for it, if you decide not to touch that in this PR and want to keep this a Grid search value.

Reply:
What exactly do you mean with avoiding, you mean the dependency?

Reply:
I avoided allowing grid search on nested structures, yes. We'll need to add the filtering to the experiment number calculation, too.